### PR TITLE
Refactor tests with uvu, improve linting

### LINKED ISSUE #16: test: rewrite and refactor to use uvu test runner
Rewrite tests to use **[uvu](https://github.com/lukeed/uvu)** test runner instead of **jest**.
The provided `uvu/assert` assertions should be enough for our purposes, but it's worth looking for alternatives.

NB. Avoid pointless duplication. Use helpers, these can be used as a base for most cases:
https://github.com/norskeld/sigma/blob/e76249b9a92c7a9218a29e062ec1d6cb9743045d/tests/%40helpers/index.ts#L39-L55

### PR DESCRIPTION
**Pull request by TheFedaikin**
Fixes #16.

Decided against doing any jest at all, going full uvu.

### CODE REVIEW

**Comment by TheFedaikin** [comment]
Should be good to go
Waiting for feedback

**Review by norskeld — CHANGES_REQUESTED** [review]
LGTM! Thanks a lot man, I appreciate the efforts.

The last thing would be to remove the `tests` directory and `paths` field from the main `tsconfig.json`.

File: .nycrc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "all": true,
+  "include": "src/**/*.ts",
+  "exclude": [
+    "src/__tests__/**/*.ts"
+  ],
+  "reporter": [
+    "text",
+    "lcov"
+  ]
+}
```
[the remark this answers is not in the document]
Comment (TheFedaikin) [reply .nycrc:14]: It should be done by default.

**Comment by TheFedaikin** [comment]
Should I also do eslint-plugin-imports to standardize the way we import things?

File: .nycrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (norskeld) [reply .nycrc:14]: Yeah, you are right. I'm blind apparently. 😄

**Comment by TheFedaikin** [comment]
Fixed some minor things along the way, should be good to go now.

**Review by norskeld — APPROVED** [review]
LGTM!

**Pull request merged**